src/components/LinterProvider/index.spec.tsx

```diff
@@ -47,11 +48,19 @@ describe(__filename, () => {
       ...moreProps,
     };
 
+    const shallowOptions = createContextWithFakeRouter();
+
     return shallowUntilTarget(
       <LinterProvider {...props}>{children}</LinterProvider>,
       LinterProviderBase,
       {
-        shallowOptions: { context: { store } },
+        shallowOptions: {
```

Comment:
This fixes the test to work with the router, but do we also want a test that proves that the bug this patch fixes is in fact fixed? Is it possible to write a test for that?

Reply:
I think it would be tricky to write a test for this given our current tools and testing approach. We would need a higher level functional test to cover it and we don't have a framework for that yet.

I agree that a test would be good but the Redux API is very stable so I think it's OK to rely on this fix via configuration. 

The risks for regression here are if someone removes the configuration without knowing it. I had originally put a comment above `withRouter()` to prevent someone from removing. You made me think that we need another comment above the `ownProps` definition, too, since removing that would also break the configuration. I just added a comment there.

Reply:
I tend to agree that writing an integration test for that would be tricky.

In other news, I filed https://github.com/mozilla/addons-code-manager/issues/585 for removing more type hints!